viewing_party/party.py

```diff
@@ -0,0 +1,147 @@
+# Two general questions:
+# 1. Should I remove these wordy docstrings? They were written to help me in 
+#    my process, but seem a bit long for "real life" (?).  
+# 2. Should I avoid list comprehensions b/c they're too confusing for 
+#    someone reading your code who is not you? (See first line of 
+#    find_user_and_friend_movies() function as an example.)
+
+def create_movie(title, genre, rating):
+    '''returns a dict with 3 key-value pairs (1 per param)'''
+    if title and genre and rating:
+        return {
+            'title': title,
+            'genre': genre,
+            'rating': rating
+        } 
+    return None 
+
+def add_to_watched(user_data, movie):
+    '''takes in a user_data dict with key 'watched' and a value which is a 
+    list of dicts repping movies the user watched and a movie to add to 
+    user_data; returns updated user_data'''
+    user_data['watched'].append(movie) 
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''takes in a user_data dict with key 'watchlist' and a value which is a 
+    list of dicts repping movies the user wants to watch and a movie to add to 
+    user_data; returns updated user_data'''
+    user_data['watchlist'].append(movie) 
+    return user_data
+
+def watch_movie(user_data, title):
+    '''takes in a user_data dict with two keys--watchlist and watched--that 
+    each correspond to a list of movies, and a title; if title in watchlist,
+    it's removed from watchlist + added to watched; returns user_data'''
+    for movie in user_data['watchlist']:
+        if title == movie['title']:
+            user_data['watchlist'].remove(movie)
+            add_to_watched(user_data, movie)
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    '''takes in a user_data dict with key 'watched' and values of list of 
+    watched movies; returns average rating of all movies'''
+    rating_sum = 0
+    num_of_movies = len(user_data['watched']) 
+    for movie in user_data['watched']:
+        rating_sum += movie['rating']
+    return 0.0 if num_of_movies == 0 else rating_sum/num_of_movies
+
+def get_most_watched_genre(user_data):
+    '''takes in a user_data dict with key 'watched' and values of list of 
+    watched movies; returns which genre occurs most frequently in list'''
+    if user_data['watched'] == []: return None 
```

Comment:
just like above! this works, but not standard practice in some python styling guides